contracts/ERC20Safe.sol

```diff
@@ -244,23 +244,29 @@ contract ERC20Safe is Initializable, BridgeRole, Pausable {
       @notice Deposit initial supply for an ESDT token already deployed on MultiversX
       @param tokenAddress Address of the contract for the ERC20 token that will be deposited
       @param amount number of tokens that need to be deposited
-\   */
+    */
     function initSupply(address tokenAddress, uint256 amount) external onlyAdmin {
         require(whitelistedTokens[tokenAddress], "Unsupported token");
+        require(!_isTokenMintBurn(tokenAddress), "Cannot init for mintable/burnable tokens");
+        require(nativeTokens[tokenAddress], "Only native tokens can be stored!");
 
-        if (!_isTokenMintBurn(tokenAddress)) {
-            totalBalances[tokenAddress] += amount;
-            IERC20 erc20 = IERC20(tokenAddress);
-            erc20.safeTransferFrom(msg.sender, address(this), amount);
-            return;
-        }
+        totalBalances[tokenAddress] += amount;
+        IERC20 erc20 = IERC20(tokenAddress);
+        erc20.safeTransferFrom(msg.sender, address(this), amount);
+    }
 
-        if (!nativeTokens[tokenAddress]) {
-            require(mintBalances[tokenAddress] >= burnBalances[tokenAddress] + amount, "Not enough minted tokens");
-        }
-        burnBalances[tokenAddress] += amount;
-        IBurnableERC20 burnableErc20 = IBurnableERC20(tokenAddress);
-        burnableErc20.burnFrom(msg.sender, amount);
+    /**
+      @notice Set burn and mint balances for a mintable/burnable token
+      @param tokenAddress Address of the contract for the ERC20 token that will be deposited
+      @param burnAmount number of tokens that are already burned
+      @param mintAmount number of tokens that are already minted
+    */
+    function initSupplyMintBurn(address tokenAddress, uint256 burnAmount, uint256 mintAmount) external onlyAdmin {
```

Comment:
for consistency, we might switch the parameters as seen here: https://github.com/multiversx/mx-bridge-eth-sc-rs/pull/202
```rust
fn init_supply_mint_burn(&self, token_id: TokenIdentifier, mint_amount: BigUint, burn_amount: BigUint) 
```